### ISSUE TITLE
No nice way to examine error when fetching user timeline for protected accounts

### ISSUE THREAD

**Issue by ekawatani** [PLhery/node-twitter-api-v2#145]
I found out that Twitter returns 200 if you try to fetch user timeline of a protected account that you do not have access to.

Using Postman, I tried `https://api.twitter.com/2/users/<id>/tweets`, using OAuth 2.0 bearer token, with V2 API and I get the following with a 200 status code:
```
{
    "errors": [
        {
            "resource_id": "<id>",
            "parameter": "id",
            "resource_type": "user",
            "section": "data",
            "title": "Authorization Error",
            "value": "<id>",
            "detail": "Sorry, you are not authorized to see the user with id: [<id>].",
            "type": "https://api.twitter.com/2/problems/not-authorized-for-resource"
        }
    ]
}
```

It's odd that Twitter returns a 200 here, but it seems like this is [by design](https://twittercommunity.com/t/users-by-username-returns-200-status-for-suspended-accounts/141550). (The linked thread discusses a case with _suspended_ account, so I also asked another question regarding a _protected_ account [here](https://twittercommunity.com/t/twitter-returns-200-when-fetching-timeline-for-a-protected-account/164450) just in case, but it's most likely by design.)

The issue I'd like to report here is that since it's not treated as an error, there's currently no nice way to retrieve this info using this library. If you investigate the response for `const response = client.v2.readOnly.userTimeline("<id>")`, then this error object is set under a property called `response._realData.errors`, but we don't have a TypeScript definition to access this. We need a better way to access this info for this case so it's not hidden under a property marked as internal with an underscore.

**Comment by alkihis** [comment]
Hi,

I agree with you, the error must be accessible more easily.
You already have access to the `_realData` property with a public accessor: `.data` (that isn't documented, I owe you this point), so you can access the error with `<paginator>.data.errors`.

This doesn't solve the root cause: This is *theorically* an error, but it isn't flagged as one.
What should we do? 
- Throw an error if the response object only contains `errors` property in case of v2 API usage?
- Add a `.errors` getter on the paginator instance (+ a possible `.hasError` boolean getter)?

**Comment by ekawatani** [comment]
> You already have access to the _realData property with a public accessor: .data

Oops, I guess I missed that. As long as we have a way to access that via the typed definitions, I don't have much to complain.

> This doesn't solve the root cause: This is theorically an error, but it isn't flagged as one. What should we do?

When working with the response, it crashes when I try to access properties of `response.meta` such as `next_token`. That's how I originally found out about this issue, but this is now expected because there was no data. And calling `fetchNext` ends in a similar error because it tries to access its properties. So, it's not really in a usable condition, but since Twitter decided to make this not an error, I'm tempted to keep the same behavior. Perhaps. we should fix it to not crash or let it crash and document it.

**Comment by alkihis** [comment]
> Perhaps. we should fix it to not crash or let it crash but document it. It's really up to you and what others think. 

I've choose to give access to errors through a `.errors` getter, and add a `.unusable` getter telling if paginator will throw if you try to get more data.

Before accessing `.meta` when the paginator prepares a new request, a assert check is made if instance is usable or not, bringing to user a meaningful error message if not: `Unable to use this paginator to fetch more data, as it does not contain any metadata. Check .errors property for more details.`

**Comment by alkihis** [comment]
Update: Fix is available in 1.9.0 :)
